lib/msal-angular/src/msal.guard.config.ts

```diff
@@ -3,10 +3,16 @@
  * Licensed under the MIT License.
  */
 
+import { RouterStateSnapshot } from "@angular/router";
 import { PopupRequest, RedirectRequest,InteractionType } from "@azure/msal-browser";
+import { Observable } from "rxjs";
+import { MsalService } from "./msal.service";
+
+export declare type MsalGuardAuthRequest = Partial<PopupRequest> | Partial<Omit<RedirectRequest, "redirectStartPage">>;
 
 export type MsalGuardConfiguration = {
     interactionType: InteractionType.Popup | InteractionType.Redirect;
-    authRequest?: Partial<PopupRequest> | Partial<Omit<RedirectRequest, "redirectStartPage">>;
+    authRequest?: MsalGuardAuthRequest | ((authService: MsalService, state: RouterStateSnapshot) => MsalGuardAuthRequest);
     loginFailedRoute?: string;
+    canActivate?: (msalService: MsalService, state?: RouterStateSnapshot) => Observable<boolean> | boolean;
```

Comment:
I'm not sure I understand why this is necessary? It seems like this would allow someone to implement their own guard logic, in which case, we would rather an app implement a custom guard, instead of using the MSAL Guard (which is intended to be generic, and will not work for all use cases).

Reply:
Furthermore, this feels like scope creep for this PR. I would prefer it is scoped to just the dynamic auth request, and have further enhacements be discussed/implemented separately.